be/src/runtime/data_stream_recvr.cc

```diff
@@ -205,6 +216,42 @@ bool DataStreamRecvr::SenderQueue::is_finished() const {
     return _is_cancelled || (_num_remaining_senders == 0 && _chunk_queue.empty());
 }
 
+bool DataStreamRecvr::SenderQueue::has_chunk() {
+    std::unique_lock<std::mutex> l(_lock);
+    if (_is_cancelled) {
```

Comment:
why when _is_cancelled is true，has_chunk is true?

Reply:
Because returned value of has_chunk is to guide computation, true means compute thread could remove this source, false means there is datas need to be accepted. And in non-pipeline, returned value of get_chunk is not used.

Reply:
> true when cancelled means compute thread could remove this source, false means there is datas need to be accepted.

So I think your return true or false is reverse.